### PR TITLE
Use the fast_scram full library

### PR DESCRIPTION
**Pull request by NelsonVides**
This simply pretends to use the extended SCRAM library that I've implemented in esl/fast_scram#3, in MIM, as well as the escalus library that already uses this.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[9008.1](https://travis-ci.org/esl/MongooseIM/jobs/753968579) / Erlang 23.0.3 / small_tests / 83182330e42dcf3e516078b3e73ff3749db7ef9d
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/3003/9008/small_tests.23.0.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/3003/9008/small_tests.23.0.3/small/ct_run.mongooseim@localhost.2021-01-11_16.44.44/lib.mongooseim.logs/run.2021-01-11_16.44.45/suite.log.html)

**Review by janciesla8818 — APPROVED** [review]
Good stuff! 🔥 

As we've rewritten the `cyrsasl_scram` module twice or three times already, what do you think of removing or updating the copyright to ESL?

**Comment by mongoose-im** [comment]
[9012.7](https://travis-ci.org/esl/MongooseIM/jobs/754159713) / Erlang 23.0.3 / elasticsearch_and_cassandra_mnesia / 1e948c0f376a04abede60cdb8b259f4ae78f9e61
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/3003/9012/elasticsearch_and_cassandra_mnesia.23.0.3)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/3003/9012/elasticsearch_and_cassandra_mnesia.23.0.3/big/ct_run.test%40travis-job-868c3684-dda1-48ec-b150-de9b4c206302.2021-01-12_18.42.21/index.html)
**OK: 330** / Failed: 0 / **User-skipped: 38** / Auto-skipped: 0

File: src/sasl/cyrsasl_scram.erl [hunk 1]
```diff
@@ -1,28 +1,8 @@
...
-%%%----------------------------------------------------------------------
-
+%%%=============================================================================
+%%% @copyright (C) 1999-2021, Erlang Solutions Ltd
+%%% @doc SASL SCRAM implementation
+%%% @end
+%%%=============================================================================
 -module(cyrsasl_scram).
 
 -author('stephen.roettger@googlemail.com').
```
Comment (janciesla8818) [inline src/sasl/cyrsasl_scram.erl:8]: With that I guess we could safely change the author as well.

Comment (NelsonVides) [reply src/sasl/cyrsasl_scram.erl:8]: Forgot xD will do later 😅

**Pull request merged**